File: rest-common/src/main/java/org/chronopolis/rest/models/storage/ActiveToggle.java
package org.chronopolis.rest.models.storage;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Objects;

/**
 * Toggle for StagingStorage active flag of a Bag
 *
 * @author shake
 */
public class ActiveToggle {
    private final boolean active;

    @JsonCreator
    public ActiveToggle(@JsonProperty("active") boolean active) {
        this.active = active;
    }

    public boolean isActive() {
        return active;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ActiveToggle toggle = (ActiveToggle) o;
        return active == toggle.active;
    }

    @Override
    public int hashCode() {

        return Objects.hash(active);
    }
}

File: rest-models/src/main/kotlin/org/chronopolis/rest/models/update/ActiveToggle.kt
package org.chronopolis.rest.models.update

/**
 * Toggle for StagingStorage to show it is no longer in use
 *
 * Note: it would be nice to have the val be 'isActive', but this causes deserialization issues. We
 * could write a deserializer for it - simple since it is only a single value, or just provide
 * 'isActive' as a method here since it is easy to impl. Not a big deal either way, for now just
 * going to add the method as it is less work (don't need to worry about registration, less classes)
 *
 * @author shake
 */
data class ActiveToggle(val active: Boolean) {
    fun isActive() = active
}
